plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -172,25 +173,17 @@ public Collection<Object> createComponents(
     LocalClusterState.state().setClusterService(clusterService);
     LocalClusterState.state().setPluginSettings((OpenSearchSettings) pluginSettings);
 
-    this.applicationContext = new AnnotationConfigApplicationContext();
-    SecurityAccess.doPrivileged(
-        () -> {
-          applicationContext.registerBean(ClusterService.class, () -> clusterService);
-          applicationContext.registerBean(NodeClient.class, () -> (NodeClient) client);
-          applicationContext.registerBean(
-              org.opensearch.sql.common.setting.Settings.class, () -> pluginSettings);
-          applicationContext.registerBean(
-              DataSourceService.class, () -> dataSourceService);
-          applicationContext.register(OpenSearchPluginConfig.class);
-          applicationContext.register(PPLServiceConfig.class);
-          applicationContext.register(SQLServiceConfig.class);
-          applicationContext.refresh();
-          return null;
-        });
+    ModulesBuilder modules = new ModulesBuilder();
+    modules.add(
+        new OpenSearchPluginModule((NodeClient) client, pluginSettings, dataSourceService));
+    Injector injector = modules.createInjector();
+    PPLService pplService =
+        SecurityAccess.doPrivileged(() -> injector.getInstance(PPLService.class));
+    this.sqlService = SecurityAccess.doPrivileged(() -> injector.getInstance(SQLService.class));
 
     // return objects used by Guice to inject dependencies for e.g.,
     // transport action handler constructors
-    return ImmutableList.of(applicationContext);
+    return ImmutableList.of(pplService);
```

Comment:
After this change, `sqlService` and `pplService` are created only once and used across all requests?

Reply:
Good points.
1. Construct PPLService in TransportAction per request.
2. Construct SQLService in RestAction per request.